### PR TITLE
Custom error on mockReject

### PR DESCRIPTION
**Pull request by alexxmde**
I'm sending this PR because I saw the need to throw custom errors on mockReject, and as it was a simple detail, I implemented it. I also added a test to the new functionality and removed a line that I imagine to be unecessary in index.js

### CODE REVIEW

File: src/.tern-port [hunk 1]
```diff
@@ -0,0 +1 @@
+40845
```
Comment (Si1kIfY) [inline src/.tern-port]: Can you remove these files?

File: src/__tests__/.tern-port [hunk 2]
```diff
@@ -0,0 +1 @@
+35009
```
Comment (Si1kIfY) [inline src/__tests__/.tern-port]: This too

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/**
+p/**
```
Comment (Si1kIfY) [inline src/index.js]: Oh man I do this all the time.... can you remove the 'p'

**Comment by jefflau** [comment]
Fixed here: https://github.com/jefflau/jest-fetch-mock/pull/40

**Pull request closed without merging**